_posts/2016-09-21-approaching-tdd-outside-android.md

```diff
@@ -0,0 +1,159 @@
+
+## But, we broke that rule
+
+It's worth mentioning that we broke this constraint somewhat to add another public method to the Account class. We did this to in order to attach the view to the account object. We needed to do so because we’ve chosen to use Android activities, and as we all know and suffer, they are instantiated by the system. Therefore, we can not pass the view through the BankAccount class constructor.
+
+We could have avoided adding this method by creating some kind of presenter, using the Account object there instead of calling it from the Activity. We consider it would make the code more complex and as it’s just a kata we’ve decided not to do so.
+
+## Extracting the acceptance criteria
+Let's start by describing the best-case workflow for tackling this problem. The starting point should ideally be a user story defining what needs to be done, who are we building it for, and why we are building it. For further information about how to write users stories, visit this <a href="https://sprint.ly/blog/agile-user-stories/?utm_content=buffer2dda2">link</a>
+
+The starting point should be a <a href="http://dannorth.net/whats-in-a-story/">user story</a> defining: What needs to be done, who are we building it for and why we are building it.
+
+The Bank Kata problem description above talks about three features so there should be three user stories. For the purpose of this post, we are going to focus on the show statement feature. The user story for the show statement could be written as follows:
+
+<b>Story: Show account statement</b>  
+
+> As a user  
+> I want to be able to show a transactions details statement  
+> So that I can easily check my account balance at any given time  
+
+With the user story completed, we can now define the acceptance criteria - the series of results that required in order for the feature to be considered done.
+
+That been said, once we got the user story, the next step would be to extract the conditions that the software must satisfy to be accepted, known as the <a href="http://www.seguetech.com/blog/2013/03/25/characteristics-good-agile-acceptance-criteria">acceptance criteria</a>. Those acceptance criteria will define a series of results that must be validated to consider that the feature is done.
+
+<a name="acceptance-criteria"></a>
+
+The acceptance criteria that we have came up with for the <b>“Show account statement”</b> story are:
+
+<b>Scenario 1: Account with transactions</b>
+
+<b>Given</b> the account has the following transactions:
+
+* A deposit of 1000 on 01/04/2014
+* A withdraw of 100 on 02/04/2014
+* A deposit of 500 on 10/04/2014
+
+<b>When</b> the user shows the account statement
+
+<b>Then</b> the statement should be a list with all the transactions in reverse chronological order
```

Comment:
Then the statement should contain a list of all the transactions in reverse chronological order.